src/pdf/ShoppingList/databaseFetch.ts

```diff
@@ -0,0 +1,43 @@
+import supabase, { Schema } from "@/supabase";
+
+export const fetchParcels = async (parcelID: string): Promise<Schema["parcels"]> => {
+    const { data, error } = await supabase.from("parcels").select().eq("primary_key", parcelID);
+    if (error !== null) {
+        throw Error(`${error.code}: ${error.message}`);
+    }
+    if (data.length !== 1) {
+        const errorMessage =
+            (data.length === 0 ? "No " : "Multiple ") + "records match this parcel ID.";
```

Comment:
template string may be better
`` `${data.length === 0 ? "No" : "Multiple"} records...` ``